### PR TITLE
NPM publish improvments and fixes

### PR DESCRIPTION
**Pull request by anxolin**
This PR addresses several issues with this project.

The main problem to address is that this project is mainly creating the strapi API, so its a strapi project. It has its own package.json with dependencies with strapi and so on.

I wanted to make a lightweight library that includes ONLY the types (and the client) to interact with the API. Well, I didn't want it at first, but @alfetopito talked me into it.

This NPM package for the lib needs a different package.json, and a different README (the strapi one don't work for this, as it carries all dependencies, and the readme focuses on running strapi). 

Solving this problem could be done with:
- Monorepo: Overkill, really, monorepo just for this is a bit annoying
- Some small hack: This PR :) 

## The hack
When you build the library, it will output the files to `lib/build` instead of `build`

That `lib` folder contains already a package.json and README (for the library) and will be the root for publishing the artifact.

To handle the publishing I did a script that will:
- build the library
- Update the `lib/package.json` with the version from `package.json` (it has a placeholder which is replaced with the right version)
- Publish the package
- Cleanup the `lib/package.json` so it's back at having the placeholder instead of the new version.  

## Bonus
As a bonus, and as the original goal of this PR before I tried to fix the above. It fixes the NPM publishing, as it was broken. 

https://github.com/cowprotocol/cms/actions/runs/8726781994/job/23942819843

You can see the script in action here:

image

## NPM artifact
As a result, you can see the artifact that was published here:
https://www.npmjs.com/package/@cowprotocol/cms/v/0.1.0-rc.18

As you can see, the readme is not the one of the main project, but is the one of the library :)

### CODE REVIEW

**Review by alfetopito — CHANGES_REQUESTED** [review]
The .npmignore might be too aggressive.

Take a look of what is in the package https://www.npmjs.com/package/@cowprotocol/cms/v/0.1.0-rc.18?activeTab=code

image

Only the readme and the package.json.
None of the files referenced on package.json are there

image

File: README.md [hunk 1]
```diff
@@ -87,3 +87,16 @@ yarn strapi version
...
 ```
+
+# Library
+This project also exposes a library that can be used to interact with the CMS API.
+
+To build the library, run:
+
+```bash
+# Create a new version (patch, minor or major)
+npm version minor
```
[the remark this answers is not in the document]
Comment (anxolin) [reply README.md:98]: This is intended. Look at the current version of that package.json :) 

https://github.com/cowprotocol/cms/pull/16/files#diff-1e946220773aef913945326261b7ee8d08b8ec29ccc66ef7c348950439212ffbR3


This is why I always reset this file https://github.com/cowprotocol/cms/pull/16/files#diff-fa994157f1b25ea2a976e6e277ab028259afd8332247e9f9cbd1f7096278a82eR7

**Comment by anxolin** [comment]
> The .npmignore might be too aggressive.
> Take a look of what is in the package https://www.npmjs.com/package/@cowprotocol/cms/v/0.1.0-rc.18?activeTab=code

I think is related to Github Actions process, because RC18 was generated there, but RC17 I generated locally and it includes the files. Will investigate what happened 

https://www.npmjs.com/package/@cowprotocol/cms/v/0.1.0-rc.17?activeTab=code

**Review by alfetopito — APPROVED** [review]
👏 

(one tiny issue in the readme)

**Pull request merged**